api/v1alpha1/httproute_types.go

```diff
@@ -93,7 +93,9 @@ const (
 
 // HeaderType constants.
 const (
-	HeaderTypeExact = "Exact"
+	HeaderTypeExact                = "Exact"
+	HeaderTypeRegularExpression    = "RegularExpression"
+	HeaderTypeImplementionSpecific = "ImplementationSpecific"
```

Comment:
When you specify a single constant for `ImplementationSpecific` matches, what do you do if an implementation has multiple different implementation-specific match types? (e.g. it might have 9 different regex options)